app/models/thread.php

```diff
@@ -44,34 +55,47 @@ public function create(Comment $comment)
         $db->commit();
     }
 
-    public static function get($id)            
-    {
-        $db = DB::conn();
 
-        $row = $db->row('SELECT * FROM thread WHERE id = ?', array($id));
+    public function update($threadId)
+    {
+        if(!$this->validate()) {
+            throw new ValidationException("Invalid Input");
+        }
+        try {
+            $db = DB::conn();
+            $params = array(
+                'title'   => $this->new_title,
+                'body'    => $this->new_thread_body
+            );
+            $db->update('thread', $params, array('id' => $threadId));
 
-        if(!$row) {
-            throw new RecordNotFoundException('no record found');
+        } catch(Exception $e) {
+            throw $e;
         }
 
-        return new self($row);                    
     }
 
-    public function getComments()
+    public static function deleteThreadById($thread_id)
     {
```

Comment:
To avoid redundancy, you may rename this function to ```deleteById($thread_id)``` since you're already on the thread model.